File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/build_execution_deployment/RequiredPlugin.java
package com.github.alexandrecarlton.idea.settings.layout.build_execution_deployment;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableRequiredPlugin.class)
public interface RequiredPlugin {
  String plugin();

  Optional<String> minimumVersion();

  Optional<String> maximumVersion();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/build_execution_deployment/RequiredPlugin.kt
package com.github.alexandrecarlton.idea.settings.layout.build_execution_deployment

import com.fasterxml.jackson.annotation.JsonProperty

data class RequiredPlugin(

    @JsonProperty("Plugin")
    val plugin: String,

    @JsonProperty("Minimum version")
    val minimumVersion: String? = null,

    @JsonProperty("Maximum version")
    val maximumVersion: String? = null
)
